### ISSUE TITLE
Schema changes

### ISSUE THREAD

**Issue by michaelcfanning** [Microsoft/sarif-sdk#1140]
1) mark result,message as required
2) order rule.configuration.defaultLevel and result.level enum values so that all shared members exist at the same ordinal value. this will allow configuration default level to be directly castable to result level in the SDK.

**Comment by michaelcfanning** [comment]
@lgolding fyi

**Comment by lgolding** [comment]
@michaelcfanning <span>#</span>1 is covered by #1138.

As for <span>#</span>2, it makes me uncomfortable to build into the SDK an assumption about the format that might not always be true. When our great-grand-editors insert a new value into one of those enumerations that breaks the correspondence,  SDK code -- or even client code! -- that performs that cast will break mysteriously. Even if our great-grand-SDK-maintainers understand this dependency and account for the change, they won't be able to fix client code.

**Comment by michaelcfanning** [comment]
Yes. We should and will do something more bullet-proof in the SDK. I would still prefer to order these similarly in the schema. 

… it may be that we need to simply separate these concepts as you suggested we consider in an offline call.

**Comment by lgolding** [comment]
Yes, the separation of concepts might help. But even if we kept the existing enumerations, and even if we adjusted the ordering as you suggest, I'd never write SDK code that relied on it, nor recommend that anyone write client code that relied on it.

The rationale for the current ordering is that it's more or less in increasing order of "severity" (if you consider "pass" to be a really low severity, and "notApplicable" even lower :-)). The order to (for example)

```
[ "note", "warning", "error", "pass", "notApplicable", "open" ]
```

seems at first glance to be random (IMO), but it does add weight to the argument that we're really mixing two concepts here.

**Comment by michaelcfanning** [comment]
As mentioned, I agree with your position on the SDK. I yield. I don't care enough about level ordering to argue the point. Let's just proceed with adding result.message as required.

**Comment by lgolding** [comment]
Ok, closing this issue, then, as `result.message` is covered by oasis-tcs/sarif-spec#283.

**Comment by michaelcfanning** [comment]
Let's leave this issue open until the SARIF SDK schema is fixed. I suggest that we leave the OASIS issue open until the OASIS copy is fixed. I realize this creates some issue duplication.
